spec/services/case_self_assign_service_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+describe CaseSelfAssignService, type: :service do
+  let(:team) { find_or_create :team_branston }
+  let(:unassigned_case) { find_or_create :offender_sar_complaint }
```

Comment:
Given that most things are passed in to this service as dependencies - I think we could do more mocking here possibly rather than actually hit the DB? What do you think? 🤔

Reply:
Dunno! I've just tried to follow the same pattern as similar specs here and this is how their setup works - don't have an opinion about it, really. If anything, I'm worried there's _too much_ mocking in this test, actually, it's the main one I used to drive out the feature